includes/abstracts/class-llms-rest-controller.php

```diff
@@ -113,4 +113,97 @@ public function get_item( $request ) {
 
 	}
 
+	/**
+	 * Retrieves the query params for retrieving a single resource.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	public function get_get_item_params() {
+
+		return array(
+			'context' => $this->get_context_param(
+				array(
+					'default' => 'view',
+				)
+			),
+		);
+
+	}
+
+	/**
+	 * Retrieve arguments for deleting a resource.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	public function get_delete_item_args() {
+		return array();
+	}
+
+	/**
+	 * Register routes.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\d]+)',
```

Comment:
Even if I globally agree on this, we have to keep in mind that, at the moment, we have at least one resource that doesn't respect this "pattern": the students enrollments, for which we have something like:
`'/' . $this->rest_base . '/(?P<post_id>[\d]+)',`

with 'args' below as:
```php
'args'   => array(
	'post_id' => array(
			'description' => __( 'Unique course, lesson, or section Identifer. The WordPress Post ID.', 'lifterlms' ),
			'type'        => 'integer',
	),
),
```
Also the WP_REST_Server::CREATABLE method's route is not `'/' . $this->rest_base,` but still `'/' . $this->rest_base . '/(?P<post_id>[\d]+)',`.

That's not a big deal, this resource will completely override the register_routes method, we know enrollments are "special".

Reply:
Ah yea, you're right. I think having this generally work for *most* resources is okay and then completely overload on the specials should be alright